src/compile/mark/init.ts

```diff
@@ -29,14 +31,25 @@ export function initMarkDef(mark: Mark | MarkDef, encoding: Encoding<string>, sc
 /**
  * Initialize encoding's value with some special default values
  */
-export function initEncoding(mark: Mark, encoding: Encoding<string>, stacked: StackProperties, config: Config): Encoding<string> {
+export function initEncoding(mark: Mark, encoding: Encoding<string>, stacked: StackProperties, config: Config, scales: Dict<Scale>): Encoding<string> {
   const opacityConfig = getMarkConfig('opacity', mark, config);
   if (!encoding.opacity && opacityConfig === undefined) {
     const opacity = defaultOpacity(mark, encoding, stacked);
     if (opacity !== undefined) {
       encoding.opacity = {value: opacity};
     }
   }
+
+  for (const channel in scales) {
```

Comment:
As mentioned in person, we may be able to do this in `normalize` because `scale.type = 'utc'` is always explicit, so we can just read fieldDef.scale inside `normalize`.